File: 800/2139B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n, m;
        cin >> n >> m;

        vector<long long> a(n);
        for (int i = 0; i < n; i++)
            cin >> a[i];

        // Sort in descending order
        sort(a.begin(), a.end(), greater<long long>());

        long long sum = 0;
        for (int i = 0; i < min(n, m); i++)
            sum += a[i] * (m - i);

        cout << sum << "\n";
    }

    return 0;
}

File: 800/2139B/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
 
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int m = sc.nextInt();
 
            Long a[] = new Long[n];
            for(int i=0;i<n;i++)
                a[i] = sc.nextLong();
 
            Arrays.sort(a, Collections.reverseOrder());
 
            long sum = 0;
            for(int i=0;i<Math.min(n, m);i++)
                sum += a[i]*(m-i);
 
            System.out.println(sum);
        }
    }
}